### ISSUE TITLE
yaml configured option default is not applied for help command

### ISSUE THREAD

**Issue by pkschweiger** [consolidation/Robo#607]
### Steps to reproduce

```
vendor/bin/robo --version
Robo 1.1.1
```

```yaml
# robo.yml
command: { hello: { options: { world: yaml } } }
```

```php
<?php
class RoboFile extends \Robo\Tasks
{
    public function hello(array $opts = ['world' => 'php'])
    {
        $this->say($opts['world']);
    }
}
```

### Expected behavior

Option default displayed by help command should be the default used when command is run.

### Actual behavior

Since ```ConfigForCommand::injectConfiguration()``` only runs in response to ```console.command``` event:
* running command uses the yaml configured default
* help displays the default from reflection

```
$ vendor/bin/robo hello
➜  yaml <-------------------------------------------+
                                                    |
$ vendor/bin/robo hello --help                      |
Options:                                            V
      --world[=WORLD]                   [default: "php"]
```

### Suggestion

Affect injection of configured defaults at the time (before) the ```option``` hook is run instead. As proof of concept the expected behavior can achieved by means of an ```option``` hook:

```php
<?php
use Consolidation\AnnotatedCommand\AnnotatedCommand;
use Consolidation\AnnotatedCommand\AnnotationData;
use Consolidation\Config\Util\ConfigFallback;

class RoboFile extends \Robo\Tasks
{
    public function hello(array $opts = ['world' => 'php''])
    {
        $this->say($opts['world']);
    }

    /**
     * Ripping off ConfigForCommand::injectConfiguration()
     *
     * @hook option
     */
    public function injectOptionDefaults(AnnotatedCommand $cmd, AnnotationData $data)
    {
        $container   = \Robo\Robo::getContainer();
        $cmdName     = str_replace(':', '.', $cmd->getName());
        $config      = $container->get('config');
        $configGroup = new ConfigFallback($config, $cmdName, 'command.', '.options.');

        foreach ($cmd->getDefinition()->getOptions() as $option => $inputOption) {
            $key   = str_replace('.', '-', $option);
            $value = $configGroup->get($key);
            if ($value !== null) {
                $inputOption->setDefault($value);
            }
        }
    }
}
```

```
$ vendor/bin/robo hello --help
Options:
      --world[=WORLD]                   [default: "yaml"]
```

**Comment by greg-1-anderson** [comment]
Part of me is wondering if users are going to care whether the default value comes from the yaml file, or built-in code. But maybe not.

Your proof-of-concept looks good. Thanks for the suggested implementation.

**Comment by pkschweiger** [comment]
My two meaningless ¢ in favor of current behavior:
* command authors code a sensible default for the general use case
* a few users/environments configure a default appropriate to theirs

Meaningless because the configured value's precedence is how Robo works.  I wouldn't change the behavior if disliked it or didn't care and saw 12 votes for coded default becoming a hard one.  Somebody out there has come to expect it from ~1.1 - as we would have yesterday (silently so if it weren't for the help quirk).

**Comment by aheimlich** [comment]
You could also do what `Consolidation\AnnotatedCommand\Options\AlterOptionsCommandEvent` does [to deal with this very issue](https://github.com/consolidation/annotated-command/blob/2a67d08447acc0006c09b8542021c87f120a32d8/src/Options/AlterOptionsCommandEvent.php#L50).

**Comment by greg-1-anderson** [comment]
@pkschweiger @aheimlich Perhaps [ConfigForCommand in consolidation/config](https://github.com/consolidation/config/blob/1.0.0/src/Inject/ConfigForCommand.php). It's even [injected into Robo's container](https://github.com/consolidation/Robo/blob/1.1.1/src/Robo.php#L212) and [added to the event dispatcher](https://github.com/consolidation/Robo/blob/1.1.1/src/Robo.php#L220).

I guess the next question is, why isn't this working as expected? The POC above should not be necessary.

**Comment by greg-1-anderson** [comment]
I guess the existing hook only runs when the command itself is executed. It also needs to run when `help` is executed.

**Comment by greg-1-anderson** [comment]
Fixed in #608
